### PR TITLE
Implement Gtk.ComboBox replacement for GTK 4

### PR DESCRIPTION
**Pull request by mathiascode**
The first step in getting rid of Gtk.CellRenderer-based widgets. They are deprecated and will be removed in GTK 5.

Remaining replacements needed in the future:
- Gtk.TreeView -> Gtk.ColumnView
- Gtk.EntryCompletion -> ?

### CODE REVIEW

**Comment by slook** [comment]
On startup with Search Files tab active on load, all comboboxes are visible, whereas some should be hidden on Global scope.

**Comment by mathiascode** [comment]
> * In Filter entry without the list shown, the up/down arrow keyboard key re-orders the history that results in two key-presses being required to get the next value, and the order is non-sequitur. Suggest not allowing up/down key binding, or use down key to open dropdown the list instead of scrolling entry in-place.

The current code related to refiltering is unfinished (I've been pulling my hair out for the past few hours trying to get it to play nicely with GTK 4 dropdowns).

**Comment by slook** [comment]
> I've been pulling my hair out

Kudos to you for the effort. However frankly I don't understand fully what brings about the urgency on this matter.

**Comment by mathiascode** [comment]
> > I've been pulling my hair out
> 
> Kudos to you for the effort. However frankly I don't understand fully what brings about the urgency on this matter?
> 
> In regards the substantial search module refactoring which are proposed in this PR, would it not be more sensible to defer this part of these changes until #2464 is completed and merged first?

There's no urgency, but the new widget restores screen reader support, which Gtk.ComboBox, Gtk.TreeView and Gtk.EntryCompletion lack in GTK 4. I initially decided to look into the new Gtk.DropDown widget, which eventually led to replacing Gtk.ComboBox entirely. Might as well get it over with while I'm motivated.

I believe everything should be working well in both GTK 3 and GTK 4 now, aside from eventual cosmetical issues in the code.

File: pynicotine/gtkgui/search.py [hunk 1]
```diff
@@ -385,6 +406,35 @@ def __init__(self, searches, text, token, mode, mode_label, show_page):
         self.selected_users = {}
         self.selected_results = {}
 
+        # Combo boxes
```
[the remark this answers is not in the document]
Comment (mathiascode) [reply pynicotine/gtkgui/search.py:409]: But we're creating the comboboxes here

**Pull request merged**